Automattic/wp-calypso
# Plans: add one-click personal plan upgrade nudge

This PR is part of #7329 and adds an upgrade nudge that allows users with a domain and a free plan to upgrade to the Personal Plan using a stored credit card. All the major functionality is here in this PR, but the component needs additional polish. I want to get this in as a WIP, and follow up with smaller PRs.

Functionality wise:
- Clicking on the 'x' will close the nudge for a page load
- Clicking on "Upgrade now" will charge your credit card, upgrade your site to a Personal Plan, and redirect you to the checkout thank you page
- Any errors during the transaction should display as a global error notice
- Clicking on the credit card info, will take you to checkout with a personal plan in cart. This should allow users to choose a different CC

<img width="753" alt="screen shot 2016-09-07 at 3 12 12 pm" src="https://cloud.githubusercontent.com/assets/1270189/18330679/dbdc79bc-750e-11e6-866d-694781cb0627.png">

A few things to note:
- Discounts are still incorrect from domains-only to Personal Plan upgrade. This should be fixed by D2633-code
- I've updated the signature in `client/lib/upgrades/actions/checkout` to return errors _and_ data on the complete callback
- We use a temporary cart to handle the upgrade. If you click on the upgrade button, don't forget to cancel your sub and refund yourself.
- Default styling on the devdocs page will override styles on this component

## Testing Instructions
- Verify that we have no functional changes, since we have no usages yet

Testing is a bit tricky, we'll need the following things:
- A site with a _free_ plan, but has a _paid_ domain
- Make sure your user has a stored credit card
- Set this site as your primary site
- Clear out your Calypso persisted state. Chome dev tools > Application > IndexedDB > Calypso
- Visit http://calypso.localhost:3000/devdocs/blocks/domain-to-plan-nudge
- Clicking on the button should take us to the checkout thank you page
- This _will_ charge your stored CC. If you want to avoid this, comment out the `submitTransaction` call, and directly call the success handler instead.

cc @rralian @lamosty @artpi @retrofox @obenland 

Test live: https://calypso.live/?branch=add/one-click-upgrade
